openwisp_controller/subnet_division/tasks.py

```diff
@@ -0,0 +1,116 @@
+import logging
+
+from celery import shared_task
+from django.db import transaction
+from django.utils.translation import ugettext_lazy as _
+from swapper import load_model
+
+logger = logging.getLogger(__name__)
+
+Subnet = load_model('openwisp_ipam', 'Subnet')
+IpAddress = load_model('openwisp_ipam', 'IpAddress')
+SubnetDivisionRule = load_model('subnet_division', 'SubnetDivisionRule')
+SubnetDivisionIndex = load_model('subnet_division', 'SubnetDivisionIndex')
+
+
+@shared_task
+def update_subnet_division_index(rule_id):
+    try:
+        division_rule = SubnetDivisionRule.objects.get(id=rule_id)
+    except SubnetDivisionRule.DoesNotExist as e:
+        logger.warning(
+            'Failed to update indexes for Subnet Division Rule '
+            f'with id: "{rule_id}", reason: {e}'
+        )
+        return
+
+    index_queryset = division_rule.subnetdivisionindex_set.all()
+    for index in index_queryset:
+        identifiers = index.keyword.split('_')
+        identifiers[0] = division_rule.label
+        index.keyword = '_'.join(identifiers)
+
+    transaction.on_commit(
```

Comment:
did I ask you to add `transaction.on_commit` here? I think it's not needed here.. what other save operation is performed in this task that we have to wait for?

Reply:
No, you did not. I will revert the changes. 👍🏼